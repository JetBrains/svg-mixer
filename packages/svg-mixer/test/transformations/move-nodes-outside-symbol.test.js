const { moveNodesOutsideSymbol } = require('svg-mixer/lib/transformations');

const { testPostSvgPlugin } = require('svg-mixer-test');

const t = testPostSvgPlugin(moveNodesOutsideSymbol);

it('Move all gradients, patterns, filters, clipMask & mask outside symbol', async () => {
  expect(await t(
    undefined,
    `<svg>
        <symbol>
            <defs>
                <linearGradient></linearGradient>
                <radialGradient></radialGradient>
                <pattern></pattern>
                <filter></filter>
                <clipPath></clipPath>
                <mask></mask>
              </defs>
        </symbol>
    </svg>`
  )).toMatchSnapshot();
});
