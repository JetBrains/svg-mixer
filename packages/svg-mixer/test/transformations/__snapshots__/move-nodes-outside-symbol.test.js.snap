// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move all gradients, patterns, filters, clipMask & mask outside symbol 1`] = `
"<svg>
  <symbol>
    <defs>






    </defs>
  </symbol>
  <linearGradient></linearGradient>
  <radialGradient></radialGradient>
  <pattern></pattern>
  <filter></filter>
  <clipPath></clipPath>
  <mask></mask>
</svg>"
`;
